drivers/lora/sx12xx_common.c

```diff
@@ -50,24 +55,33 @@ int __sx12xx_configure_pin(const struct device * *dev, const char *controller,
 static void sx12xx_ev_rx_done(uint8_t *payload, uint16_t size, int16_t rssi,
 			      int8_t snr)
 {
-	Radio.Sleep();
-
 	dev_data.rx_buf = payload;
 	dev_data.rx_len = size;
 	dev_data.rssi = rssi;
 	dev_data.snr = snr;
 
+	if (MODEM_FREE(dev_data)) {
```

Comment:
Would it be better to do this in sx12xx_lora_recv instead? I.e., just setup the buffer here and give the semaphore and call sleep when we take the semaphore in recv. IMO, that would make the code a bit cleaner.

Reply:
That is an alternative, controlling the state in the callback will still need to be done for the TX case though, so that would make the logic different between TX and RX.

Reply:
Good point. Let's keep it symmetric in that case.